### ISSUE TITLE
Mesh deselection - ArgumentNullException

### ISSUE THREAD

**Issue by Drexen** [LogicalError/realtime-CSG-for-unity#168]
Not entirely sure what causes this but happens often when deselecting objects. I think when pressing escape to clear my selection.

```
ArgumentNullException: Argument cannot be null.
Parameter name: sourceArray

System.Array.Copy (System.Array sourceArray, System.Array destinationArray, Int32 length) (at /Users/builduser/buildslave/mono/build/mcs/class/corlib/System/Array.cs:931)
RealtimeCSG.MeshEditBrushTool.DeselectAll () (at Assets/Plugins/RealtimeCSG/Editor/Scripts/EditorWindows/BrushTools/MeshEditBrushTool.cs:331)
RealtimeCSG.CSGBrushEditorManager.DeselectAll () (at Assets/Plugins/RealtimeCSG/Editor/Scripts/EditorWindows/BrushEditorManager/CSGBrushEditorManager.cs:223)
RealtimeCSG.SelectionUtility.DeselectAll () (at Assets/Plugins/RealtimeCSG/Editor/Scripts/Helpers/SelectionUtility.cs:113)
RealtimeCSG.Keys.HandleSceneKeyUp (IBrushTool tool, Boolean checkForTextEditing) (at Assets/Plugins/RealtimeCSG/Editor/Scripts/Settings/Keys.cs:185)
RealtimeCSG.UnitySceneEventHandler.OnScene (UnityEditor.SceneView sceneView) (at Assets/Plugins/RealtimeCSG/Editor/Scripts/Managers/UnitySceneEventHandler.cs:889)
UnityEditor.SceneView.CallOnSceneGUI () (at C:/buildslave/unity/build/Editor/Mono/SceneView/SceneView.cs:2075)
UnityEditor.SceneView.HandleSelectionAndOnSceneGUI () (at C:/buildslave/unity/build/Editor/Mono/SceneView/SceneView.cs:1405)
UnityEditor.SceneView.OnGUI () (at C:/buildslave/unity/build/Editor/Mono/SceneView/SceneView.cs:1242)
System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at /Users/builduser/buildslave/mono/build/mcs/class/corlib/System.Reflection/MonoMethod.cs:222)

**Comment by LogicalError** [comment]
I'll look into it .. it sounds familiar so I might've already fixed this in the next version I'm working on, I'll check.

**Comment by LogicalError** [comment]
As far as I can see this should've been fixed in 1.358, which is available from the update site:
https://prenominal.nl/updates/
When this version is available from the asset store I'll close this ticket
